src/test/java/com/google/step/youtube/YouTubeInfoScraperTest.java

```diff
@@ -310,4 +313,57 @@ public void scrapeVideoIdsFromSearch_IOException() throws IOException {
+    }
+
+    /* Multiple items with one item having no promo-code in the description. */
+    @Test
+    public void scrapePromoCodesFromVideos_multipleItemsSomeCodesFound() throws IOException {
+        VideoListResponse testVideoResponse = new VideoListResponse();
+        // Two company names one promo-code found.
+        String snippet = "DISCOUNT!!! For a limited time only, take 25% off your first purchase at " + KEYWORD
+                + " with the code OVDOINGTHINGS25. This offer is valid online only. ";
+        String description1 = snippet + "\nMust apply code at " + KEYWORD + " checkout. Expires August 31, 2020.";
+        // One keyword two promo-codes found.
+        String description2 = "And if you want to order food through " + KEYWORD + " go to "
+                + "https://pmfleet.app.link/zyoaw9s3R6 and use my code A1JZN";
+        String descriptionWithNoCode = KEYWORD + " is a great company!";
+        testVideoResponse.setItems(Arrays.asList(
+                new Video().setId(VIDEO_ID)
+                        .setSnippet(new VideoSnippet().setTitle(VIDEO_TITLE).setDescription(description1)
+                                .setPublishedAt(new DateTime(DATE))),
+                new Video().setId(VIDEO_ID)
+                        .setSnippet(new VideoSnippet().setTitle(VIDEO_TITLE).setDescription(description2)
+                                .setPublishedAt(new DateTime(DATE))),
+                new Video().setId(VIDEO_ID).setSnippet(new VideoSnippet().setTitle(VIDEO_TITLE)
+                        .setDescription(descriptionWithNoCode).setPublishedAt(new DateTime(DATE)))));
+        when(mockListVideos.execute()).thenReturn(testVideoResponse);
+        Optional<List<PromoCode>> actual = scraper.scrapePromoCodesFromVideos(KEYWORD, VIDEO_ID_LIST);
+        assertThat(actual.get(), equalTo(Arrays.asList(
+                PromoCode.create("OVDOINGTHINGS25", snippet, VIDEO_ID, VIDEO_TITLE, DATE),
+                PromoCode.create("A1JZN", description2, VIDEO_ID, VIDEO_TITLE, DATE),
+                PromoCode.create("https://pmfleet.app.link/zyoaw9s3R6", description2, VIDEO_ID, VIDEO_TITLE, DATE))));
+    }
+
+    @Test
+    public void scrapePromoCodesFromVideos_oneItemNoCodesFound() throws IOException {
+        VideoListResponse testVideoResponse = new VideoListResponse();
+        String descriptionWithNoCode = KEYWORD + " is a great company!";
+        testVideoResponse.setItems(Arrays.asList(new Video().setId(VIDEO_ID).setSnippet(new VideoSnippet()
+                .setTitle(VIDEO_TITLE).setDescription(descriptionWithNoCode).setPublishedAt(new DateTime(DATE)))));
+        when(mockListVideos.execute()).thenReturn(testVideoResponse);
+        Optional<List<PromoCode>> actual = scraper.scrapePromoCodesFromVideos(KEYWORD, VIDEO_ID_LIST);
+        assertThat(actual.get().isEmpty(), equalTo(true));
+    }
+
+    @Test
+    public void scrapePromoCodesFromVideos_IOException() throws IOException {
```

Comment:
Is there any special logic when an IO exception is thrown that's supposed to be tested here? It looks like you're just testing that the IOException from the mock is propagated and nothing else.

Reply:
No logic is being tested, just trying to increase test coverage.  Should I remove it?

Reply:
I don't think it tests any branches in your own code so it doesn't really increase the coverage, so feel free to remove it.

Reply:
Removed.